src/mito2/src/memtable/merge_tree/data.rs

```diff
@@ -218,26 +365,58 @@ impl DataBuffer {
     }
 }
 
+#[derive(Debug)]
 pub(crate) struct DataBufferIter {
     batch: RecordBatch,
     offset: usize,
-    current_pk_index: PkIndex,
+    current_data_batch: Option<DataBatch>,
 }
 
-impl Iterator for DataBufferIter {
-    type Item = Result<DataBatch>;
+impl DataBufferIter {
+    pub(crate) fn new(batch: RecordBatch) -> Self {
+        let mut iter = Self {
+            batch,
+            offset: 0,
+            current_data_batch: None,
+        };
+        iter.next(); // fill data batch for comparison and merge.
```

Comment:
`next()` returns a result. We should handle it or change the return value of `next()`.